src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs

```diff
@@ -442,11 +442,13 @@ internal static void ParseAuthorityIfNecessary(MergedOptions mergedOptions)
         {
             if (string.IsNullOrEmpty(mergedOptions.TenantId) && string.IsNullOrEmpty(mergedOptions.Instance) && !string.IsNullOrEmpty(mergedOptions.Authority))
             {
-                const int StartingIndex = 8; // length of "https://"
+                ReadOnlySpan<char> doubleSlash = "//".AsSpan();
                 ReadOnlySpan<char> authoritySpan = mergedOptions.Authority.AsSpan().TrimEnd('/');
+                int startingIndex = authoritySpan.IndexOf(doubleSlash) == -1 ? 0 : authoritySpan.IndexOf(doubleSlash) + doubleSlash.Length;

```

Comment:
Call IndexOf() once.

```suggestion
                int doubleSlashIndex = authoritySpan.IndexOf(doubleSlash)
                int startingIndex = doubleSlashIndex == -1 ? 0 : doubleSlashIndex + doubleSlash.Length;
```

Reply:
good point thanks